src/firebolt/async_db/_types.py

```diff
@@ -190,6 +191,64 @@ def parse_value(
     raise DataError(f"Unsupported data type returned: {ctype.__name__}")
 
 
-def format_value(value: ColType) -> str:
-    # TODO: FIR-7793
-    pass
+escape_chars = {
+    "\0": "\\0",
+    "\\": "\\\\",
+    "'": "\\'",
+}
+
+
+def format_value(value: ParameterType) -> str:
+    """For python value to be used in a SQL query"""
+    if isinstance(value, bool):
+        return str(int(value))
+    if isinstance(value, (int, float)):
+        return str(value)
+    elif isinstance(value, str):
+        return f"'{''.join(escape_chars.get(c, c) for c in value)}'"
+    elif isinstance(value, datetime):
+        if value.tzinfo is not None:
+            value = value.astimezone(timezone.utc)
+        return f"'{value.strftime('%Y-%m-%d %H:%M:%S')}'"
+    elif isinstance(value, date):
+        return f"'{value.isoformat()}'"
+    if value is None:
+        return "NULL"
+    elif isinstance(value, Sequence):
+        return f"[{', '.join(format_value(it) for it in value)}]"
+
+    raise DataError(f"unsupported parameter type {type(value)}")
+
+
+def format_sql(query: str, parameters: Sequence[ParameterType]) -> str:
```

Comment:
Just thinking, could this be done with a `re.sub()`? So there's less re-inventing the wheel, as I feel like the problem of replacing the placeholders is not new.